edc-extensions/data-encryption/src/test/java/org/eclipse/tractusx/edc/data/encryption/TxEncryptorExtensionTest.java

```diff
@@ -49,7 +48,7 @@ class TxEncryptorExtensionTest {
     @BeforeEach
     void setup(ObjectFactory factory, ServiceExtensionContext c) {
         c.registerService(Monitor.class, monitor);
-        context = spy(c);
+        context = c;
```

Comment:
do we need the `context` at all? we can inject it directly as method arg if needed.

Reply:
Yeah i didn't want to change the tests so much, probably it's not needed, this was only done because tests where failing :). I would tackle all this other refactor in another PR. I guess there are others as well